src/plots/waterfallPlot.ts

```diff
@@ -0,0 +1,219 @@
+     */
+    public total<T>(): Plots.AccessorScaleBinding<T, boolean>;
+    /**
+     * Sets total to a constant number or the result of an Accessor
+     * 
+     * @param {Accessor<boolean>}
+     * @returns {Plots.Waterfall} The calling Waterfall Plot.
+     */
+    public total(total: Accessor<boolean>): Waterfall<X, Y>;
+    public total(total?: Accessor<boolean>): any {
+      if (total == null) {
+        return this._propertyBindings.get(Waterfall._TOTAL_KEY);
+      }
+      this._bindProperty(Waterfall._TOTAL_KEY, total, null);
+      return this;
+    }
+
+    protected _additionalPaint(time: number) {
+      this._connectorArea.selectAll("line").remove();
+      if (this._connectorsEnabled) {
+        Utils.Window.setTimeout(() => this._drawConnectors(), time);
+      }
+    }
+
+    protected _createNodesForDataset(dataset: Dataset) {
+      var drawer = super._createNodesForDataset(dataset);
+      this._connectorArea = this._renderArea.append("g").classed(Waterfall._CONNECTOR_AREA_CLASS, true);
+      return drawer;
+    }
+
+    protected _extentsForProperty(attr: string) {
+      var primaryAttr = "y";
+      if (attr === primaryAttr) {
+        return [this._extent];
+      } else {
+        return super._extentsForProperty(attr);
+      }
+    }
+
+    protected _generateAttrToProjector() {
+      var attrToProjector = super._generateAttrToProjector();
+
+      var yScale = this.y().scale;
+      var totalAccessor = Plot._scaledAccessor(this.total());
+
+      var yAttr = this.attr("y");
+      if (yAttr == null) {
+        attrToProjector["y"] = (d, i, dataset) => {
+          var currentValue = this.y().accessor(d, i, dataset);
+          var isTotal = totalAccessor(d, i, dataset);
+          if (isTotal) {
+            return Math.min(yScale.scale(currentValue), yScale.scale(0));
+          } else {
+            var currentSubtotal = this._subtotals[i];
+            if (i === 0) {
+              if (currentValue < 0) {
+                return yScale.scale(currentSubtotal - currentValue);
+              } else {
+                return yScale.scale(currentSubtotal);
+              }
+            }
+            var priorSubtotal = this._subtotals[i - 1];
+            if (currentSubtotal > priorSubtotal) {
+              return yScale.scale(currentSubtotal);
+            } else {
+              return yScale.scale(priorSubtotal);
+            }
+          }
+        };
+      }
+
+      var heightAttr = this.attr("height");
+      if (heightAttr == null) {
+        attrToProjector["height"] = (d, i, dataset) => {
+          var isTotal = totalAccessor(d, i, dataset);
+          var currentValue = this.y().accessor(d, i, dataset);
+          if (isTotal) {
+            return Math.abs(yScale.scale(currentValue) - yScale.scale(0));
+          } else {
+            var currentSubtotal = this._subtotals[i];
+            if (i === 0) {
+              return Math.abs(yScale.scale(Math.abs(currentValue)) - yScale.scale(0));
```

Comment:
Hmm, what happens when the first value is not 0-based (say the entire waterfall plot starts from value 100). 

Say the first delta is 100-200. Will this have height 200? 

To my understanding the height attr should be the height of the bar (so a delta from 100 to 200 should have height 100?